### PR TITLE
EVG-13722: db auth framework

### PR DESCRIPTION
**Pull request by hhoke**

### CODE REVIEW

**Comment by bsamek** [comment]
I'd like to take a look at this before merging. Adding @ybrill as well. (I'd like to be cautious about these changes, since they could cause the app not to start.)

**Comment by ybrill** [comment]
Given the overlap with https://github.com/evergreen-ci/evergreen/pull/4361 my plan is to hold off on reviewing this PR until that review closes out.
@hhoke let me know if it would be helpful for you to arrange this differently (like if you wanted this in place to let you set up staging cedar, for instance)

**Comment by hhoke** [comment]
@ybrill I want to run this first, and test on cedar staging, before the evergreen PR. I have tested this with the authless version here, in evergreen, and have tested this with an auth file on my mac, using a local mongodb with auth enabled.

File: operations/flags.go
Comment (hhoke) [reply operations/flags.go:126]: Default should be the zero value for now. Otherwise we might accidentally break prod if we don't set the environment variable to "".

Comment (ybrill) [reply operations/flags.go:126]: Why would it break? 
I would think it would look for the file, not find it, grip.Error, and keep going

Comment (hhoke) [reply operations/flags.go:126]: Right now there's a unified chef environment for cedar staging and cedar prod, so the yaml files will show up on prod once we turn that on. We'll have to run prod without auth and staging with auth for some time. Once we migrate cedar prod db to atlas, I think it's fine to set this default to our yaml file.

Comment (ybrill) [reply operations/flags.go:126]: ah I get it now.
still, we can omit setting `Value:  ""`: it's implied.

File: operations/setup.go
Comment (ybrill) [reply operations/setup.go:165]: would you mind to remove the comment?

**Review by ybrill — APPROVED** [review]

File: environment.go
Comment (julianedwards) [inline environment.go:67]: Nit: remove blank line.

File: environment.go
Comment (julianedwards) [inline environment.go:61]: Nit: remove blank line.

**Review by julianedwards — APPROVED** [review]
LGTM modulo small nits

File: operations/setup.go
Comment (bsamek) [inline operations/setup.go:149]: nit: It seems incorrect to log the error in this case but not in the case that YAML errors. I would suggest not logging at all and instead relying on the caller.

File: operations/setup.go
Comment (bsamek) [inline operations/setup.go:168]: nit: delete this empty line

File: operations/setup.go
Comment (bsamek) [inline operations/setup.go:169]: I think this can be `creds, err :=`. I think it's unnecessary to initialize those two variables separately.

File: operations/setup.go
Comment (bsamek) [inline operations/setup.go:170]: nit: You don't need an if statement here. You can just `grip.Error(err)`. If err is not nil, this will noop.

**Review by bsamek — CHANGES_REQUESTED** [review]
just nits

File: operations/setup.go
Comment (hhoke) [reply operations/setup.go:149]: Good catch, this was cruft from a time of extremely indiscriminate debug logging

File: operations/setup.go
Comment (hhoke) [reply operations/setup.go:169]: Creds needs to be initialized outside the if loop or they go out of scope and can't be used in return. If I don't initialize err and use a walrus, it creates a shadowed creds. As written is the only way I was able to get build to work.

**Review by bsamek — APPROVED** [review]

**Pull request merged**
